### ISSUE TITLE
Is your inception resnet v2 different from the original paper?

### ISSUE THREAD

**Issue by yxchng** [soeaver/caffe-model#7]
I was inspecting your model and I think that your inception resnet v2 is not the same as the paper. Is it that I am misreading or that it is really different from the paper? 

https://github.com/soeaver/caffe-model/blob/master/inception_resnet.py

The stem is not the same at all. The paper has two branches but your is only one branch. Am I right?

**Comment by yxchng** [comment]
also no average pooling is used in the paper if I am not wrong

**Comment by soeaver** [comment]
Indeed, the inception resnet v2 implemented by tensorflow is slightly different with the paper. My model is converted from the tensorflow version,but you get the basically same accuracy as the paper.

**Comment by yxchng** [comment]
This is a screenshot of part of the tensorflow model
<img width="521" alt="screen shot 2017-03-22 at 11 40 37 am" src="https://cloud.githubusercontent.com/assets/10518587/24181198/7c401e52-0ef4-11e7-881d-393f031f9bf0.png">

Did you also add a lot of batch norm when the tensorflow model doesn't?

**Comment by soeaver** [comment]
I am not very familiar with the operation of tensorflow, but it seems like the batch norm layer is added as follow?
![image](https://cloud.githubusercontent.com/assets/8629203/24181653/571bfd6e-0ef7-11e7-9143-7c3095489224.png)

**Comment by yxchng** [comment]
Ok. I miss out that part. Thanks a lot. One last question (hope that I am not bothering you) - you don't have the following scaling part in your caffe code right?

<img width="434" alt="screen shot 2017-03-22 at 12 52 55 pm" src="https://cloud.githubusercontent.com/assets/10518587/24182714/7c718faa-0efe-11e7-9eeb-279fcbe68d1c.png">

I am intending to train the model from scratch.

**Comment by soeaver** [comment]
![image](https://cloud.githubusercontent.com/assets/8629203/24182939/073b3ef0-0f00-11e7-9130-209a43f930d5.png)
coeff: 1, 0.17 is the scaling parameter.

In my experience, training any inception model from scratch is very difficult, I sincerely recommend you just finetune my model with your dataset. 
You can download from BaiduCloud https://pan.baidu.com/s/1jHPJCX4#list/path=%2F

**Comment by yxchng** [comment]
Can I know what is the hard part in training inception model from scratch? Do people just keep restarting the training after it dies?

**Comment by soeaver** [comment]
Inception model is very hard to optimize, even you follow all the detail as the google's papers say. I don't know why the optimization is so difficult, you can search some relevant repositories to learning some others' experience.

**Comment by yxchng** [comment]
Hi I see that your prototxt has a batch size of 64. My Titan X Pascal can only support up to 8 and it is already using more than 7gb. Is that normal?

**Comment by soeaver** [comment]
This is normal on caffe, inception-resnet-v2 is memory exhaustion. 64 is just a placeholder when I wrote the script.

**Comment by yxchng** [comment]
You say "on caffe". Do you imply that caffe is memory inefficient?

**Comment by soeaver** [comment]
Relative to mxnet, it's yes.

**Comment by yxchng** [comment]
Hi, just checking. How long did you train your inception resnet model? I am using one pascal titan x and based on my calculation, it is going to take ~6 months for 200 epochs. Is it normal?

**Comment by soeaver** [comment]
@yxchng 
I just finetuned inception-resnet-v2 for my custom dataset. I forget the specific time consumption, but I remember it's really time-consuming. So taking ~6 months for 200 epoch of imagenet on a single pascal titan x may be normal.
